articles/sql-or-nosql-when-to-choose-what/index.md

```diff
@@ -0,0 +1,123 @@
+
+### SQL or NoSQL (high-level differences)
+
+#### Relation
+
+SQL employs a kind of relationship that exists between two or more database tables. As we have explained above, NoSQL is non-relational. To get this point correct, think of two towns let's say town A and town B. Take town A, where the residents understand only one language. They use it to do all sorts of communication and interaction around them day in day out. To change the language of this town will bring a lot of confusion and disruptions among the residents. Take town B, the residents speak different languages. Everyone communicates and interacts with the world around them differently. There is no universal way to understand each other. When a new resident is introduced to this town, the language of this new resident will affect no one.
+
+Town A represents the relational SQL, and Town B represents the non-relational NoSQL. SQL is powerful and versatile. On the other hand, they are restrictive and structured with a predefined schema. If you choose to use them, your data must use and follow the same structure. And as in Town A, changing its structure would be difficult. It will disrupt your whole system. The language of SQL is tabular with rows and columns, you can only follow that structure.
+
+NoSQL on the hand, Just like town B, it has a dynamic unstructured schema that is flexible to any new resident. Its data can be stored in many ways. This data can be
+- Document-oriented
+- Column-oriented
+- Graph base or
+- Key-value store.
+
+It is extremely flexible
+- Each document you add has no defined structure
+- Each document has its own distributed structure
+- To add files in a document as you go
+
+#### Schema
+
+Schema is a blueprint of how data is organized in a database. These two database models have a completely different database schema representation. To understand this further check out this joke.
+
+![image title](\database-admins.jpg)
+
+If these admins walked into an SQL bar, they would have defiantly found a table to enjoy their drink from. This means NoSQL has no tabular schema and SQL does.
+
+#### Scalability
+
+Assume and figure out a building near your neighbourhood. Be the architecture and check the possibility of adding more floors to the existing building to host more rooms. Would you propose adding more floors to the existing building or would you build a new one? If you will add more floors, what are the chances of this building to remain stable? That the case here. SQL is scaled vertically, which means that a server capacity can be increased by increasing specks such as, CPU, RAM and disk size (HDD or SDD) (more floors will be added to the existing building). Vertical scaling adds more power to your existing server. The downside of that of course is that there will be a limit point hit on how much power you can stretch your server. That is one of the restrictions of a SQL database approach that makes scaling to be super hard.
```

Comment:
**Assume and figure out a building near your neighbourhood.**

"Think of a building in your neighbourhood" flows better.

Reply:
**Be the architecture** 

"Be the Architect" is the correct term and would be more suitable.